app/src/main/java/com/b07project2024/group1/CatalogItem.java

```diff
@@ -86,6 +86,14 @@ public boolean equals(Object obj) {
                     && description.equals(other.description)
                     && period.equals(other.period)
                     && videoURLs.equals(other.videoURLs);
+        } else if ((imageURLs != null && other.imageURLs != null) && (videoURLs != null && other.videoURLs != null)) {
+            return lot.equals(other.lot)
+                    && name.equals(other.name)
+                    && category.equals(other.category)
+                    && description.equals(other.description)
+                    && period.equals(other.period)
+                    && videoURLs.equals(other.videoURLs)
+                    && imageURLs.equals(other.imageURLs);
```

Comment:
There might be a way  to simplify with Objects.equals(a,b) && Objects.equals... etc. since it checks for nullity